src/Mutator/Extensions/MBString.php

```diff
@@ -0,0 +1,262 @@
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Mutator\Extensions;
+
+use Generator;
+use Infection\Mutator\Util\Mutator;
+use Infection\Mutator\Util\MutatorConfig;
+use PhpParser\Node;
+
+/**
+ * @internal
+ */
+final class MBString extends Mutator
+{
+    private const MB_CASES = [
+        'MB_CASE_UPPER' => 0,
+        'MB_CASE_LOWER' => 1,
+        'MB_CASE_TITLE' => 2,
+        'MB_CASE_FOLD' => 3,
+        'MB_CASE_UPPER_SIMPLE' => 4,
+        'MB_CASE_LOWER_SIMPLE' => 5,
+        'MB_CASE_TITLE_SIMPLE' => 6,
+        'MB_CASE_FOLD_SIMPLE' => 7,
+    ];
+
+    private $converters;
+
+    public function __construct(MutatorConfig $config)
+    {
+        parent::__construct($config);
+
+        $settings = $this->getSettings();
+
+        $this->setupConverters($settings);
+    }
+
+    /**
+     * @return Node|Node[]|Generator
+     */
+    public function mutate(Node $node)
+    {
+        yield from $this->converters[$this->getFunctionName($node)]($node);
+    }
+
+    protected function mutatesNode(Node $node): bool
+    {
+        return isset($this->converters[$this->getFunctionName($node)]);
+    }
+
+    private function setupConverters(array $functionsMap): void
+    {
+        $converters = [
+            'mb_chr' => $this->mapNameSkipArg('chr', 1),
+            'mb_ereg_match' => $this->mapEreg($this->mapNameSkipArg('preg_match', 2), '^', '', [$this, 'warpEqOne']),
```

Comment:
I added this kind of mapper. Basically it converts

```diff
- mb_ereg_match($pattern, $subject, $options);
+ preg_match('/^' . \str_replace('/', '\/', $pattern) . '/', $subject) === 1;
```

I'm not 100% sure if it will be reliable.

1. It will be hard to map `$options` argument. We can easily map `m` modifier but I do not think the rest has equivalents
2. ereg and preg regular expressions might not be compatible.

Maybe we should separate ereg family to separate mutator that is not on by default or drop support for them altogether ?

Reply:
I'm all in for the latter. Let's just ignore `mb_ereg_*` for time being, shall we?

Reply:
fine for me

Reply:
removed